### PR TITLE
[`flake8-import-conventions`] Add a rule for `BannedImportAlias`

### LINKED ISSUE #3892: Support the `banned-import-alias`
## 🚀  Feature request

Currently, `ruff` supports enforcing importing packages/modules under aliases. It would be, however, nice to have a feature to forbid some aliases which might be considered non-standard at some companies.

For example, with a hypothetical configuration
```
[tool.ruff.flake8-import-conventions.forbidden-aliases]
"tensorflow.keras.backend" = "K"
```
the following import
```
>>> import tensorflow.keras.backend as K
```

should raise an error.

Thanks a lot for all your hard work. Ruff is an amazing tool ❤️

### PR DESCRIPTION
**Pull request by stancld**
This PR adds a new rule into `flake8-import-conventions` allowing to ban of certain import aliases (`ICN002`), for example, due to violating some project's rules.

For example, with a given configuration:
```
[tool.ruff.flake8-import-conventions.banned-aliases]
"tensorflow.keras.backend" = ["K"]
```
the following import
```
>>> import tensorflow.keras.backend as K
```
raises.
```
`tensorflow.keras.backend` should not be imported as `K`
```

Closes #3892.

### Question:
- Since this rule is not supported directly in `flake8-import-conventions`, I'm not sure if adding this rule under this module is valid or not.

### CODE REVIEW

**Comment by charliermarsh** [comment]
One question on semantics. Given this:

```py
[tool.ruff.flake8-import-conventions.banned-aliases]
"tensorflow.keras.backend" = ["K"]
```

Is it correct that we'll flag `import tensorflow.keras.backend as K` but not `import tensorflow.keras.backend as B` etc.? In practice, do you all tend to alias `tensorflow.keras.backend` as something else, or prefer that it's not aliased at all? (Specifically, I'm wondering if it'd be useful to frame this as "disallow aliasing this import" rather than banning specific aliases.)

**Comment by stancld** [comment]
My rationale when implementing this was to ban specific aliases I encounter during the reviewing of a project's codebase. (For example, those are aliases regularly used in tutorials, etc., but are considered unconventional in a given project.)
Didn't think about forbidding aliases for specific imports at all, but I can see your point now, and it makes sense to me as well. To be honest, I'm not sure then whether it's more appropriate to rely on a global ban for selected imports, or whether those rules are rather complementary.

File: crates/ruff/src/rules/flake8_import_conventions/mod.rs [hunk 1]
```diff
@@ -44,6 +45,38 @@ mod tests {
...
+                            vec!["K".to_string()],
+                        ),
+                        ("torch.nn.functional".to_string(), vec!["F".to_string()]),
+                    ])),
+                }
+                .into(),
+                ..Settings::for_rule(Rule::BannedImportAlias)
+            },
+        )?;
+        assert_yaml_snapshot!("custom_banned", diagnostics);
```
[the remark this answers is not in the document]
Comment (stancld) [reply crates/ruff/src/rules/flake8_import_conventions/mod.rs]: @MichaReiser Thanks for pointing this out. Updated :]

**Comment by charliermarsh** [comment]
I'm not sure either (whether the rules are complementary), but let's just run with this for now since we have a concrete use-case.

File: crates/ruff/src/rules/flake8_import_conventions/rules/check_banned_import.rs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+pub fn check_banned_import(
+    import_from: &Stmt,
+    name: &str,
+    asname: &str,
+    banned_conventions: &FxHashMap<String, Vec<String>>,
+) -> Option<Diagnostic> {
+    if let Some(banned_aliases) = banned_conventions.get(name) {
+        if banned_aliases
+            .iter()
+            .any(|banned_alias| banned_alias == asname)
```
Comment (charliermarsh) [inline crates/ruff/src/rules/flake8_import_conventions/rules/check_banned_import.rs:51]: I tweaked the logic here a bit, I think it was unintentionally flagging any non-aliased import.

**Pull request merged**